optout/src/main/java/gov/cms/ab2d/optout/OptOutProcessing.java

```diff
@@ -0,0 +1,182 @@
+package gov.cms.ab2d.optout;
+
+
+import com.amazonaws.services.lambda.runtime.LambdaLogger;
+import com.amazonaws.services.s3.model.AmazonS3Exception;
+import gov.cms.ab2d.databasemanagement.DatabaseUtil;
+import software.amazon.awssdk.core.ResponseInputStream;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.model.GetObjectResponse;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Map;
+import java.util.Optional;
+import java.util.TreeMap;
+
+import static gov.cms.ab2d.optout.OptOutConstants.*;
+
+public class OptOutProcessing {
+
+    private final String fileName;
+    private final LambdaLogger logger;
+    public TreeMap<Long, OptOutResult> optOutResultMap;
+    private static final S3Client S3_CLIENT = S3Client.builder()
+            .region(S3_REGION)
+            .build();
+
+    public OptOutProcessing(String fileName, LambdaLogger logger) {
+        this.fileName = fileName;
+        this.logger = logger;
+        this.optOutResultMap = new TreeMap<>();
+    }
+
+    public void process() {
+        try {
+            processFileFromS3();
+            createResponseOptOutFile(createResponseContent());
+        //    deleteFileFromS3();
+        } catch (IOException ex) {
+            throw new OptOutException(ex);
+        }
+    }
+
+    public void processFileFromS3() throws IOException {
+        Connection dbConnection = DatabaseUtil.getConnection();
+        GetObjectRequest getObjectRequest = GetObjectRequest.builder()
+                .bucket(BFD_S3_BUCKET_NAME)
+                .key(fileName)
+                .build();
+
+        ResponseInputStream<GetObjectResponse> s3ObjectResponse = S3_CLIENT.getObject(getObjectRequest);
+
+        BufferedReader reader = new BufferedReader(new InputStreamReader(s3ObjectResponse));
```

Comment:
Great job on implementing the BufferedReader, lets load test to make sure it processes well and doesnt exceed optout 15 mins max processing time.

Reply:
[AB2D-5872](https://jira.cms.gov/browse/AB2D-5872) OptOut Lambda Load testing